### ISSUE TITLE
how to set import base path

### ISSUE THREAD

**Issue by DerayGa** [benmosher/eslint-plugin-import#480]
here is my condition, **how to set import baseDir?**

I put code with folder in src/
e.g.
```
src
|--classA
     |--ClassA.js
|--classB
     |--ClassB.js
```

if ClassA import ClassB, it will write like
```
const ClassB = import('../classB/ClassB');
```
---

and I also use **'gulp-flatten'** to flatten **'src/' to 'lib/'** when release.
after flatten, the folder structure is gone, and look like this.
```
lib
|--ClassA.js
|--ClassB.js
```

so the import path is also change to.
 ```
const classB = import('./ClassA');
```

which is short, not more ../../ relative path, but this will pop line import error.
because this path isn't correct when run lint under 'src/' folder.
(and it will be correct when run lint under 'lib/' folder)

TL;DR
**is there any setting/config that I can set baseDir/basePath to 'lib/' for eslint-plugin-import ?**

therefore when I write
 ```
const classB = import('./ClassB');
```
the lint check will know that it's import from $(project)/lib/ClassB.js
not matter the code in under 'src/' or 'lib/'

thank you

**Comment by benmosher** [comment]
Hi @DerayGa! You might want to muck around with either the `paths` or `moduleDirectories` settings of `node-resolve`, assuming you're using the Node resolver: https://www.npmjs.com/package/eslint-import-resolver-node

**Comment by DerayGa** [comment]
Hi Ben,
thank you for reply. :smiley:

I did try eslint-import-resolver-node(v0.2.2)
and this is my setting for resolver

```
  'settings': {
    'import/resolver': {
      node: {
        'paths': ['$(project)/lib']
      },
    },
  },
```

but still not working,
even I set the path to **ABSOLUTE** path, the import still pop 'unresoloved' error
e.g.
'/Users/Deray/awesome/lib'

did I do it the wrong way? :sweat:

**Comment by benmosher** [comment]
Ah, yeah. I'm not sure if this is doable with the stock Node resolver...

**Comment by benmosher** [comment]
You might be able to use https://www.npmjs.com/package/eslint-import-resolver-babel-module-alias? I don't use it, so I'm not sure if it can be configured (easily) for what you want.

I'm going to close for now as this is a pretty unique setup, that I don't expect to implement direct support for. I hope you're able to get something working, though!

**Comment by DerayGa** [comment]
many thanks. Ben

Hope you well
